actors/miner/tests/policy_test.rs

```diff
@@ -0,0 +1,108 @@
+use fil_actor_miner::quality_for_weight;
+use fil_actor_miner::{
+    DEAL_WEIGHT_MULTIPLIER, QUALITY_BASE_MULTIPLIER, SECTOR_QUALITY_PRECISION,
+    VERIFIED_DEAL_WEIGHT_MULTIPLIER,
+};
+use fil_actors_runtime::EPOCHS_IN_DAY;
+use fvm_shared::bigint::BigInt;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::sector::SectorSize;
+
+#[test]
+fn quality_is_independent_of_size_and_duration() {
+    // Quality of space with no deals. This doesn't depend on either the sector size or duration.
+    let empty_quality = TokenAmount::from(1 << SECTOR_QUALITY_PRECISION);
+    // Quality space filled with non-verified deals.
+    let deal_quality =
+        &empty_quality * (DEAL_WEIGHT_MULTIPLIER.clone() / QUALITY_BASE_MULTIPLIER.clone());
+    // Quality space filled with verified deals.
+    let verified_quality = &empty_quality
+        * (VERIFIED_DEAL_WEIGHT_MULTIPLIER.clone() / QUALITY_BASE_MULTIPLIER.clone());
+
+    let size_range: Vec<SectorSize> = vec![
+        SectorSize::_2KiB,
+        SectorSize::_8MiB,
+        SectorSize::_512MiB,
+        SectorSize::_32GiB,
+        SectorSize::_64GiB,
+    ];
+    let duration_range: Vec<ChainEpoch> = vec![
+        ChainEpoch::from(1),
+        ChainEpoch::from(10),
+        ChainEpoch::from(1000),
+        1000 * EPOCHS_IN_DAY,
+    ];
+
+    for size in size_range {
+        for duration in &duration_range {
+            let sector_weight = weight(size, *duration);
+            assert_equal(
+                &empty_quality,
+                &quality_for_weight(size, *duration, &BigInt::from(0), &BigInt::from(0)),
+            );
+            assert_equal(
+                &deal_quality,
+                &quality_for_weight(size, *duration, &sector_weight, &BigInt::from(0)),
+            );
+            assert_equal(
+                &verified_quality,
+                &quality_for_weight(size, *duration, &BigInt::from(0), &sector_weight),
+            );
+        }
+    }
+}
+
+#[test]
+fn quality_scales_with_verified_weight_proportion() {
+    // Quality of space with no deals. This doesn't depend on either the sector size or duration.
+    let empty_quality = TokenAmount::from(1 << SECTOR_QUALITY_PRECISION);
+    // Quality space filled with verified deals.
+    let verified_quality = &empty_quality
+        * (VERIFIED_DEAL_WEIGHT_MULTIPLIER.clone() / QUALITY_BASE_MULTIPLIER.clone());
+
+    let size_range: Vec<SectorSize> = vec![
+        SectorSize::_2KiB,
+        SectorSize::_8MiB,
+        SectorSize::_512MiB,
+        SectorSize::_32GiB,
+        SectorSize::_64GiB,
+    ];
+    let sector_duration = ChainEpoch::from(1_000_000); // ~350 days
+
+    let verified_range: Vec<SectorSize> = vec![
```

Comment:
Iterating over only valid sector sizes above is correct.  These are the only weight values that make sense.  Here this does not make sense.  We need the test to cover examples of verified weight that don't fully pack the sector.  What you are really iterating over here is *deal* size which is unconstrained beyond fitting into a sector. 

The way to approach this is to have a single sector size and then iterate over the same verified spaces as the go test.  You'll need to use a different type than sector size, just big int.  To make this work you'll either need to change the type signature of the local weight function or create a new weight function that takes in deal size (big int) and duration.  This won't require any change to actor code since quality_for_weight never touches sector size

Reply:
Had to create a new weight function; the compiler complained about typecasting with a generic type. I believe I've gotten the logic correct now, but let me know if I missed anything!